metricflow/plan_conversion/spec_transforms.py

```diff
@@ -88,34 +90,49 @@ class CreateOnConditionForCombiningMetrics(InstanceSpecSetTransform[SqlExpressio
     The latter scenario consolidates the rows keyed by 'c' into a single entry.
     """
 
-    # here
     def __init__(  # noqa: D
         self,
         column_association_resolver: ColumnAssociationResolver,
         table_aliases_in_coalesce: Sequence[str],
         table_alias_on_right_equality: str,
+        offset: SqlJoinTimeOffset = SqlJoinTimeOffset(),
     ) -> None:
         self._column_association_resolver = column_association_resolver
         self._table_aliases_in_coalesce = table_aliases_in_coalesce
         self._table_alias_on_right_equality = table_alias_on_right_equality
+        self._offset = offset
 
-    # here
-    def _make_equality_expr(self, column_alias: str) -> SqlExpressionNode:
-        left_expr = _make_coalesced_expr(self._table_aliases_in_coalesce, column_alias)
-        right_expr = SqlColumnReferenceExpression(
+    def _make_equality_expr(
+        self, column_alias: str, offset: SqlJoinTimeOffset = SqlJoinTimeOffset()
+    ) -> SqlExpressionNode:
+        left_expr: SqlExpressionNode = _make_coalesced_expr(self._table_aliases_in_coalesce, column_alias)
+        right_expr: SqlExpressionNode = SqlColumnReferenceExpression(
             col_ref=SqlColumnReference(
                 table_alias=self._table_alias_on_right_equality,
                 column_name=column_alias,
             )
         )
-        # if either has offset window, build date_add expr
-        # if either has offset grain to date, build date_trunc expr
+        if offset.left_offset:
```

Comment:
Sorry, this whole class no longer exists........ check out the SqlQueryPlanJoinBuilder